divi/src/masternode-payments.cpp

```diff
@@ -469,22 +465,25 @@ bool CMasternodePayments::AddWinningMasternode(const CMasternodePaymentWinner& w
         return false;
     }
 
+    CMasternodeBlockPayees* payees;
     {
         LOCK2(cs_mapMasternodeBlocks, cs_mapMasternodePayeeVotes);
 
-        if (mapMasternodePayeeVotes.count(winnerIn.GetHash())) {
+        auto* winner = GetPaymentWinnerForHash(winnerIn.GetHash());
+        if (winner == nullptr)
             return false;
-        }
 
-        mapMasternodePayeeVotes[winnerIn.GetHash()] = winnerIn;
+        *winner = winnerIn;
```

Comment:
By the above discussion this would be a nullptr when the above item is fixed

Reply:
Fixed together with the other comment.